### PR TITLE
Error notification elastic search

### PR DESCRIPTION
**Pull request by chrizzor**
This adds an error notification if elasticsearch is not running.

Original branch:
https://github.com/IkramMaalej/kitodo-production/tree/error-notification-elastic-search

### CODE REVIEW

File: Kitodo/src/main/java/de/sub/goobi/forms/HelperForm.java [hunk 1]
```diff
@@ -304,4 +305,13 @@ public String getUserAgent() {
     public boolean getUserStepDoneSearchEnabled() {
         return ConfigCore.getBooleanParameter("withUserStepDoneSearch");
     }
+    public static void saveMessage() {
+
```
Comment (henning-gerhardt) [inline Kitodo/src/main/java/de/sub/goobi/forms/HelperForm.java]: Please remove empty lines which are not needed.

File: Kitodo/src/main/java/de/sub/goobi/forms/HelperForm.java [hunk 2]
```diff
@@ -304,4 +305,13 @@ public String getUserAgent() {
     public boolean getUserStepDoneSearchEnabled() {
         return ConfigCore.getBooleanParameter("withUserStepDoneSearch");
     }
+    public static void saveMessage() {
```
Comment (henning-gerhardt) [inline Kitodo/src/main/java/de/sub/goobi/forms/HelperForm.java]: 1) Public method without JavaDoc - please add JavaDoc.
2) Why you are adding a new method to display waring / error messages? There are a lot of them already existing and can be used.

File: Kitodo/src/main/java/de/sub/goobi/forms/HelperForm.java [hunk 3]
```diff
@@ -304,4 +305,13 @@ public String getUserAgent() {
...
+
+
+        FacesContext context = FacesContext.getCurrentInstance();
+
+        context.addMessage(null, new FacesMessage(FacesMessage.SEVERITY_ERROR,"Failed",  "The elastic search server is not running!") );
```
Comment (henning-gerhardt) [inline Kitodo/src/main/java/de/sub/goobi/forms/HelperForm.java]: 1) Please fix formatting of method parameters.
2) This kind of message is not translatable - is this intended?

**Comment by henning-gerhardt** [comment]
Running this branch without any running Elastic Search server cause this following messages in Kitodo log file:

```
WARNING: There are some unhandled FacesMessages, this means not every FacesMessage had a chance to be rendered.
These unhandled FacesMessages are: 
- java.net.ConnectException: Connection refused
- java.net.ConnectException: Connection refused
- java.net.ConnectException: Connection refused
- (DataException: java.net.ConnectException: Connection refused)
- (DataException: java.net.ConnectException: Connection refused)
- (DataException: java.net.ConnectException: Connection refused)
```

Are this messages are intended?

**Comment by chrizzor** [comment]
This messages are not intended. But they were also thrown before adding the error notification.
Currently the error message appears if a request is send to the unreachable elasticsearch (like getDocument() countDocument() ...).
Maybe a central status check is needed and if the server is not reachable, no requests should be sent?

**Comment by henning-gerhardt** [comment]
Your solution sounds good but we should ask @Beacze as she implemented the elastic search component.

**Comment by Beacze** [comment]
Sounds good.

**Comment by henning-gerhardt** [comment]
Ok, then this kind of messages should be handled in a separate commit.

**Pull request merged**